### DISCUSSION TITLE
Looking for a plugins example + question (v-calendar)

### DISCUSSION THREAD

**Discussion by fosterdouglas** [nuxt/nuxt#17463] [Questions] [answered]
Hello!

I'm having some trouble getting a plugin working for [v-calendar](https://vcalendar.io/).
I'm pretty new to Nuxt and Nuxt 3 (but familiar with Vue 2 and Vue 3).

My **/plugins/v-calendar.js** file seems relatively simple:

```
import { defineNuxtPlugin } from '#app'
import DatePicker from 'v-calendar'

export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.vueApp.use(DatePicker)
})
```

and in **nuxt.config.js**, I've got:

```
import vue from '@vitejs/plugin-vue'
...
build: {
    transpile: [/v-calendar/], // i don't fully understand what transpile is doing
  },
build: {
    transpile: ['v-calendar'], // and tried both of these as I'm not sure the difference
  },
vite: {
    plugins: [
      vue({
        template: {
          compilerOptions: {
            isCustomElement: (tag) => tag.startsWith('DatePicker'), // registering the component seems to be the issue
          },
        },
      }),
    ],
}
...
```

Also tried, from seeing via many other discussions, these:

- Changing the plugin file name to **v-calendar.client.js** to render client-side only
- and then using `<ClientOnly>` tags around the`<DatePicker>` in the app

With the most common error either being an m.default screen, or:

```
Failed to resolve component: DatePicker
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement. 
```

Anybody have some advice for a plugins newbie? In general, I wish there was a bit more documentation on this outside of the VueGtag example, although I fully understand the beta nature of things currently. There's also of course the chance that I've set it up correctly and just this specific package isn't compatible right now (although it does have Vue 3 support), but I don't know exactly how I'd be able to deduce if that's the case.

Thank you!!

**Comment by wobsoriano** [comment, marked as answer]
The warning I get from your setup, which you can ignore because it's not a custom element

```
[Vue warn]: Failed to resolve component: v-calendar
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.
```

The working code:

```ts
// nuxt.config.ts
import { defineNuxtConfig } from 'nuxt3';

export default defineNuxtConfig({
  css: ['v-calendar/dist/style.css']
});
```

```ts
// plugins/v-calendar.client.ts
import { defineNuxtPlugin } from '#app';
import VCalendar from 'v-calendar';

export default defineNuxtPlugin((nuxtApp) => {
  nuxtApp.vueApp.use(VCalendar);
});
```

```html
<template>
  <div>
    <ClientOnly><v-calendar /></ClientOnly>
  </div>
</template>
```

But I assume you only need it 1-2 pages? Just import them manually on that page.

```html
<script setup lang="ts">
import { Calendar } from 'v-calendar';
</script>

<template>
  <div>
    <ClientOnly><Calendar /></ClientOnly>
  </div>
</template>
```

With that you don't have to create a plugin and you will not have any warnings.

**Reply by fosterdouglas** [reply]
Thank you @wobsoriano ! This makes more sense to me now.

**Reply by yaosint** [reply]
Hi, @wobsoriano 

The following is my issue. 

`Uncaught (in promise) RangeError: Maximum call stack size exceeded`

Please let me know what is the problem.
Thanks

**Reply by wobsoriano** [reply]
@yaosint a simple repro would be nice
